sound/soc/sof/ipc4-topology.h

```diff
@@ -435,6 +435,37 @@ struct sof_ipc4_src {
 	struct sof_ipc4_msg msg;
 };
 
+/*
+ * struct sof_ipc4_asrc_data - IPC data for ASRC
+ * @base_config: IPC base config data
+ * @sink_rate: Output rate for sink module
+ * @asrc_mode: Mask of allowed ASRC features:
+ *             - BITS 0-1 (ASRC mode):
+ *	           a. 0 1 - Playback mode / PUSH mode
+ *	           b. 1 0 - Capture mode / PULL mode
```

Comment:
Just out of curiosity:
We have:
input format(s)
output format(s)
out_rate.

If ASRC mode is 01b the _input_ is _input_ , _output_ is _output_ 
If ASRC mode is 10b the _input_ is _output_ , _output_ is _input_ 

Or what this direction switch does with an ASRC in playback or capture path?

Reply:
I'll add a better description to comment. Also there's need to fix the ASRC component to follow this init IPC. Seems the ASRC is not yet fully IPC4.